### ISSUE TITLE
Rust function must return enum

### ISSUE THREAD

**Issue by levBagryansky** [objectionary/eo#2283]
Now rust function looks like
```
pub fn foo(mut env: EOEnv<'_>) -> i32 {
      env.find("^");
      0
    }
```
So return value is the vertex of eo object. Thus In order for the insert to be valid, it must contact java several times via function from #2237(`find`, `copy` and so on). I believe iis much better to create a enum like:
```
pub enum Returning {
    EO (u32),
    Double(f64),
    Int(i64),
    String(String),
    Bytes(Box<[u8]>),
}
```
So rust functions will be:
```
pub fn foo(mut env: EOEnv<'_>) -> Returning {
    let v = env.find("aboba");
    return EO(v);
}
```
So user can return either the object or the data.
The intermediate function now looks llike 
```
#[no_mangle]
pub extern "system" fn
Java_EOrust_natives_native0_native0
<'local> (env: JNIEnv<'local>, _class: JClass<'local>, universe: JObject<'local>) -> jint
{ 
    return foo(EOEnv::new(env, _class, universe)); 
}
```
It can return byte representation of the enum instead. For example, the first byte specify type, others are the raw representation.  In java side we can decipher this message.
@yegor256 WDYT?

**Comment by yegor256** [comment]
@levBagryansky looks good to me

**Comment by 0pdd** [comment]
@levBagryansky 3 puzzles [#2313](https://github.com/objectionary/eo/issues/2313), [#2314](https://github.com/objectionary/eo/issues/2314), [#2315](https://github.com/objectionary/eo/issues/2315) are still not solved.

**Comment by 0pdd** [comment]
@levBagryansky 4 puzzles [#2313](https://github.com/objectionary/eo/issues/2313), [#2314](https://github.com/objectionary/eo/issues/2314), [#2315](https://github.com/objectionary/eo/issues/2315), [#2384](https://github.com/objectionary/eo/issues/2384) are still not solved.

**Comment by 0pdd** [comment]
@levBagryansky 3 puzzles [#2313](https://github.com/objectionary/eo/issues/2313), [#2314](https://github.com/objectionary/eo/issues/2314), [#2315](https://github.com/objectionary/eo/issues/2315) are still not solved; solved: [#2384](https://github.com/objectionary/eo/issues/2384).

**Comment by 0pdd** [comment]
@levBagryansky 2 puzzles [#2313](https://github.com/objectionary/eo/issues/2313), [#2315](https://github.com/objectionary/eo/issues/2315) are still not solved; solved: [#2314](https://github.com/objectionary/eo/issues/2314), [#2384](https://github.com/objectionary/eo/issues/2384).

**Comment by 0pdd** [comment]
@levBagryansky all 4 puzzles are solved here: [#2313](https://github.com/objectionary/eo/issues/2313), [#2314](https://github.com/objectionary/eo/issues/2314), [#2315](https://github.com/objectionary/eo/issues/2315), [#2384](https://github.com/objectionary/eo/issues/2384).
